storage-access.bs

```diff
@@ -224,9 +234,41 @@ When invoked on {{Document}} |doc|, the <dfn export method for=Document><code>re
 
 ISSUE: Shouldn't step 3.7 be [=same site=]?
 
+When invoked on {{Document}} |doc|, the <dfn export method for=Document><code>requestStorageAccessForSite(site)</code></dfn> method must run these steps:
+
+1. Let |p| be [=a new promise=].
+1. If this algorithm was invoked when |doc|'s {{Window}} object did not have [=transient activation=], [=reject=] and return |p|.
+1. If |doc|'s [=Document/browsing context=] is not a [=top-level browsing context=], [=reject=] and return |p|.
+1. If |doc|'s [=Document/origin=] is an [=opaque origin=], [=reject=] and return |p|.
+1. Let |parsedURL| be the the result of running the [=URL parser=] on |site|.
+1. If |parsedURL| is failure, [=reject=] and return |p|.
```

Comment:
Reject with a `TypeError` here (we'll change the other rejections to also throw TypeErrors or meaningful errors where possible)